### PR TITLE
ccip: chain reader router discovery function.

### PR DESCRIPTION
**Pull request by winder**
Add new function to onramp chain reader config for discovering destination router.

Misc cleanup for naming conventions.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -74,7 +74,7 @@ require (
 	github.com/shopspring/decimal v1.4.0
 	github.com/smartcontractkit/chain-selectors v1.0.23
 	github.com/smartcontractkit/chainlink-automation v1.0.4
-	github.com/smartcontractkit/chainlink-ccip v0.0.0-20240930150148-1c731b9602dd
+	github.com/smartcontractkit/chainlink-ccip v0.0.0-20240930194309-f49d68904bdf
```
Comment (winder) [inline go.mod]: Change back to main once https://github.com/smartcontractkit/chainlink-ccip/pull/171 is merged.

Comment (winder) [reply go.mod]: done.

File: core/capabilities/ccip/configs/evm/contract_reader.go [hunk 2]
```diff
@@ -198,18 +208,34 @@ var SourceReaderConfig = evmrelaytypes.ChainReaderConfig{
...
+					ChainSpecificName: mustGetMethodName("getStaticConfig", onrampABI),
+					ReadType:          evmrelaytypes.Method,
+				},
+				consts.MethodNameOnRampGetDynamicConfig: {
+					ChainSpecificName: mustGetMethodName("getDynamicConfig", onrampABI),
+					ReadType:          evmrelaytypes.Method,
+				},
+				// TODO: swap with const.
+				"OnRampGetDestChainConfig": {
+					//consts.MethodNameOffRampGetDestChainConfig: {
```
Comment (makramkd) [inline core/capabilities/ccip/configs/evm/contract_reader.go]:
```suggestion
					//consts.MethodNameOnRampGetDestChainConfig: {
```

File: core/capabilities/ccip/configs/evm/contract_reader.go [hunk 3]
```diff
@@ -113,6 +113,16 @@ var DestReaderConfig = evmrelaytypes.ChainReaderConfig{
 					ChainSpecificName: mustGetEventName(consts.EventNameExecutionStateChanged, offrampABI),
 					ReadType:          evmrelaytypes.Event,
 				},
+				//nolint:staticcheck // TODO: remove deprecated config.
+				consts.MethodNameOfframpGetStaticConfig: {
```
Comment (asoliman92) [inline core/capabilities/ccip/configs/evm/contract_reader.go:117]: This is duplicated, I see it on line 96 as well

File: core/capabilities/ccip/configs/evm/contract_reader.go [hunk 4]
```diff
@@ -113,6 +113,16 @@ var DestReaderConfig = evmrelaytypes.ChainReaderConfig{
...
+					ChainSpecificName: mustGetMethodName("getStaticConfig", offrampABI),
+					ReadType:          evmrelaytypes.Method,
+				},
+				//nolint:staticcheck // TODO: remove deprecated config.
+				consts.MethodNameOfframpGetDynamicConfig: {
```
Comment (asoliman92) [inline core/capabilities/ccip/configs/evm/contract_reader.go:122]: duplicated?

File: core/capabilities/ccip/configs/evm/contract_reader.go (hunk 3 above)
Comment (winder) [reply core/capabilities/ccip/configs/evm/contract_reader.go:117]: Most places seem to use capitol r in OffRamp/OnRamp, I'm fixing this case to follow that convention. After this change I'll update `chainlink-ccip` to use the new name, and after that change I'll remove these deprecated configs.

File: core/capabilities/ccip/configs/evm/contract_reader.go (hunk 4 above)
Comment (winder) [reply core/capabilities/ccip/configs/evm/contract_reader.go:122]: [See reply above](https://github.com/smartcontractkit/chainlink/pull/14607#discussion_r1783057179).

**Review by asoliman92 — DISMISSED** [review]

**Review by makramkd — APPROVED** [review]

**Review by asoliman92 — APPROVED** [review]

**Pull request merged**